spec/unit/algebra/shortcuts/test_cross_product.rb

```diff
@@ -0,0 +1,42 @@
+require 'spec_helper'
+module Bmg
+  module Algebra
+    describe Shortcuts, "cross_product" do
+
+      let(:left) {
+        Relation.new([
+          { a: "foo",  b: 1 },
+          { a: "bar",  b: 2 }
+        ], Type::ANY.with_attrlist([:a, :b]))
+      }
+
+      let(:right) {
+        Relation.new([
+          { c: "baz",  d: 3 },
+          { c: "zap",  d: 4 }
+        ], Type::ANY.with_attrlist([:z, :c]))
+      }
+
+      subject {
+        left.cross_product(right)
+      }
+
+      it 'compiles as expected' do
+        expect(subject).to be_a(Operator::Join)
+        expect(left_operand(subject)).to be(left)
+        expect(right_operand(subject)).to be(right)
+        expect(subject.send(:on)).to eql([])
+      end
+
+      it 'works as expected' do
+        expect(subject.to_set).to eql([
+          { a: "bar", b: 2, c: "zap", d: 4 },
+          { a: "foo", b: 1, c: "baz", d: 3 },
+          { a: "foo", b: 1, c: "zap", d: 4 },
+          { a: "bar", b: 2, c: "baz", d: 3 },
+        ].to_set)
```

Comment:
@blambeau I didn't like the fact that tests rely on the ordering imposed by `to_a`, so I'm using `to_set` on both subject and expected value.